llvm/unittests/ADT/StringExtrasTest.cpp

```diff
@@ -380,3 +382,14 @@ TEST(StringExtrasTest, arrayToStringRef) {
   roundTripTestString("\0\n");
   roundTripTestString("\xFF\xFE");
 }
+
+TEST(StringExtrasTest, isPunct) {
+  // Loop over all valid char values and verify that llvm::isPunct matched
+  // std::ispunct(). Limit to first 256 characters in case CHAR_BIT > 8.
+  int Count = 0;
+  for (char C = std::numeric_limits<char>::min(); Count < 256; ++C) {
+    EXPECT_EQ(static_cast<bool>(std::ispunct(C)), isPunct(C));
```

Comment:
`std::ispunct` result is locale-dependent and should not be used as a reference.

Reply:
What's the alternative? (I'm unfamiliar with locale and related issues). Call setlocale(LC_CTYPE, "C") at the start of this function? Or just have a curated list of punct and non punct characters?

Reply:
I guess a curated list is ok: libcxx/test/std/localization/locales/locale.convenience/classification/ispunct.pass.cpp does the same.

Reply:
> Or just have a curated list of punct and non punct characters?

Maybe just check a few? There are tests at the beginning of this file that test similar interfaces, they don't check all possible values.

I'm not very familiar with locales either, and it is possible that calling `std::ispunct` is totally safe here (or can be made safe by setting C locale).

Reply:
Done. Do just a few spot checks.